nmdc_runtime/api/endpoints/nmdcschema.py

```diff
@@ -131,6 +134,70 @@ def get_by_id(
     )
 
 
+@router.get("/nmdcschema/ids/{hypothetical_doc_id}/collection-and-class-name")
+def get_collection_names_and_class_name_by_doc_id(hypothetical_doc_id: str):
+    r"""
+    Gets the name of the Mongo collection that could contain a document having this `id`
+    and the name of the NMDC Schema class whose instances could have this `id`.
+    """
+    # Note: The `nmdc_runtime.api.core.metadata.map_id_to_collection` function is
+    #       not used here because that function (a) only processes collections whose
+    #       names end with `_set` and (b) only works for `id` values that are in
+    #       use in the database (as opposed to hypothetical `id` values).
+
+    # Extract the typecode portion, if any, of the specified `id`.
+    #
+    # Examples:
+    # - "nmdc:foo-123-456" → "foo"
+    # - "foo:nmdc-123-456" → `None`
+    #
+    pattern = re.compile(r"^nmdc:(\w+)?-")
+    match = pattern.search(hypothetical_doc_id)
+    typecode_portion = match.group(1) if match else None
+
+    if typecode_portion is None:
+        return None  # abort
+
+    # Determine the schema class, if any, of which the specified `id` could belong to an instance.
+    schema_class_name = None
+    for typecode in typecodes():
+        if typecode_portion == typecode["name"]:
```

Comment:
Thank you, @dwinston! (Maybe needless to say), this change you made fixes a bug in the code I wrote (oops).